### PR TITLE
JA4 for TLS and QUIC -- v12

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #10829

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/6379

Changes to previous PR:
* Rebase against current master.
* Use local copy of JA3 state flag to ensure consistency across multiple dependent checks.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1761

### CODE REVIEW

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #10856, thanks!

File: configure.ac [hunk 1]
```diff
@@ -2207,6 +2207,27 @@
...
+
+    AC_ARG_ENABLE(ja4,
+           AS_HELP_STRING([--disable-ja4], [Disable JA4 support]),
+           [enable_ja4="$enableval"],
+           [enable_ja4=yes])
+    if test "$enable_ja4" = "yes"; then
+        AC_DEFINE([HAVE_JA4],[1],[JA4 enabled])
+        enable_ja4="yes"
+    fi
+    AM_CONDITIONAL([HAVE_JA4], [test "x$enable_ja4" != "xno"])
```
[the remark this answers is not in the document]
Comment (satta) [reply configure.ac:2228]: Hmmm? These are enabled by default.

[the remark this answers is not in the document]
Comment (catenacyber) [reply configure.ac:2228]: ok thanks

[the remark this answers is not in the document]
Comment (satta) [reply configure.ac:2228]: That is as intended, right? At least that's how I understood it... enable them by default but make it possible to disable at build time?

[the remark this answers is not in the document]
Comment (satta) [reply configure.ac:2228]: Yep. Just confirming that we wanted it that way :)

**Pull request closed without merging**